File: NetProve/Models/GameSession.cs
using System;
using System.Collections.Generic;
using NetProve.Models;

namespace NetProve.Models
{
    public sealed class GameSession
    {
        public string GameName { get; init; } = "";
        public string Platform { get; init; } = "";
        public int ProcessId { get; init; }
        public DateTime StartTime { get; init; } = DateTime.Now;
        public DateTime? EndTime { get; set; }
        public TimeSpan Duration => (EndTime ?? DateTime.Now) - StartTime;

        // Performance samples collected during session
        public List<double> PingSamples { get; } = new();
        public List<double> JitterSamples { get; } = new();
        public List<double> PacketLossSamples { get; } = new();
        public List<float> CpuSamples { get; } = new();
        public List<float> RamSamples { get; } = new();
        public int LagSpikeCount { get; set; }
        public List<LagCause> DetectedBottlenecks { get; } = new();
        public bool IsActive => EndTime == null;
    }
}

File: android/app/src/main/java/com/netprove/app/model/GameSession.kt
package com.netprove.app.model

data class GameSession(
    val packageName: String,
    val gameName: String,
    val startTime: Long = System.currentTimeMillis(),
    var endTime: Long? = null,
    val cpuSamples: MutableList<Float> = mutableListOf(),
    val ramSamples: MutableList<Float> = mutableListOf(),
    val pingSamples: MutableList<Double> = mutableListOf(),
    val jitterSamples: MutableList<Double> = mutableListOf(),
    val packetLossSamples: MutableList<Double> = mutableListOf()
) {
    val isActive: Boolean get() = endTime == null
    val durationMs: Long get() = (endTime ?: System.currentTimeMillis()) - startTime
}
